packages/web-frontend/src/sagas.js

```diff
@@ -775,25 +749,18 @@ function* fetchCurrentMeasureInfo() {
   const selectedMeasureId = selectCurrentMeasureId(state);
 
   if (currentOrganisationUnitCode) {
-    const isHierarchyPopulated = measureHierarchy.length;
-
-    // Update the default measure ID
-    if (isHierarchyPopulated) {
-      const newMeasure = getSelectedMeasureFromHierarchy(
-        measureHierarchy,
-        selectedMeasureId,
-        selectCurrentProject(state),
-      );
-
-      if (newMeasure !== selectedMeasureId) {
-        yield put(setMeasure(newMeasure));
-      }
-    } else {
+    const isHierarchyPopulated = !!measureHierarchy.length;
```

Comment:
This whole commit is unrelated, and could/should probably be reverted to save regression. What do you think?

Reply:
Before I review this change, could you give me some context about it? Is it intended as pure refactoring?

Reply:
Yes, it should be purely refactoring (plus a small performance loss? from using selectors (because of the overhead and lack of recalculation in real life))

Reply:
The motivation was that I found the existing code very confusing.

Reply:
Fair enough! I think it is a valuable refactor. Remember to leave regression testing notes in the card

Reply:
Will do 👍